cache/blobs.go

```diff
@@ -46,121 +47,112 @@ func (sr *immutableRef) computeBlobChain(ctx context.Context, createIfNeeded boo
 func computeBlobChain(ctx context.Context, sr *immutableRef, createIfNeeded bool, compressionType compression.Type, forceCompression bool, s session.Group) error {
 	baseCtx := ctx
 	eg, ctx := errgroup.WithContext(ctx)
-	var currentDescr ocispec.Descriptor
 	if sr.parent != nil {
 		eg.Go(func() error {
 			return computeBlobChain(ctx, sr.parent, createIfNeeded, compressionType, forceCompression, s)
 		})
 	}
+
+	var descr ocispec.Descriptor
 	eg.Go(func() error {
-		dp, err := g.Do(ctx, sr.ID(), func(ctx context.Context) (interface{}, error) {
-			refInfo := sr.Info()
-			if refInfo.Blob != "" {
-				if forceCompression {
-					desc, err := sr.ociDesc()
-					if err != nil {
-						return nil, err
-					}
-					if err := ensureCompression(ctx, sr, desc, compressionType, s); err != nil {
-						return nil, err
-					}
+		refInfo := sr.Info()
```

Comment:
This doesn't look quite equivalent. Previously if differ was in progress, `!createIfNeeded` would still wait for completion. And `!forceCompression` should not call differ twice, even in mediatype is different.

Reply:
> And !forceCompression should not call differ twice, even in mediatype is different

Fixed

> Previously if differ was in progress, !createIfNeeded would still wait for completion

Agree that used to be the behavior, but was it on purpose? It seems strange to me for the behavior of `createIfNeeded` to be dependent on the value of `createIfNeeded` in other builds that happen to be running at the same time. If a user sets `createIfNeeded` to false, I feel like it makes the most sense to just never block on the differ running, especially since it's related to timing and thus never a deterministic thing anyways.

Let me know if I'm missing something, I can make it behave in the old way but it will convolute the code a bit more.